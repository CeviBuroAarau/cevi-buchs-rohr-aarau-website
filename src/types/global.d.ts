declare let global: any; // eslint-disable-line @typescript-eslint/no-explicit-any
