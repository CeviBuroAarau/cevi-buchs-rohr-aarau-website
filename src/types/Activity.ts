export interface Activity {
  title: string;
  url: string;
  thumb: string;
}

export interface CockpitActivities {
  entries: ActivityRaw[];
}

export interface ActivityRaw {
  title: string;
  image: {
    path: string;
  };
  thumb: {
    path: string;
  };
}
