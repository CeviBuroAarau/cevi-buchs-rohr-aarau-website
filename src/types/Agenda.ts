export interface Agenda {
  title: string;
  text: string;
  date: Date;
}

export interface CockpitAgenda {
  entries: Agenda[];
}

export interface CockpitScope {
  display: string;
}

export interface CockpitEventInfo {
  scope: CockpitScope;
  text: string;
  date: Date;
}

export interface EventInfo {
  scope: string;
  text: string;
  date: Date;
}

export interface CockpitEventInfos {
  entries: CockpitEventInfo[];
}
