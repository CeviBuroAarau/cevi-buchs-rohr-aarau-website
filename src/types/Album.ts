export interface Album {
  title: string;
  dateString: string;
  date: Date;
  year: number;
  images: AlbumImage[];
  previewImage: AlbumPreviewImage;
}

export interface AlbumImage {
  title: string;
  url: string;
}

export interface AlbumPreviewImage {
  url: string;
}

export interface CockpitAlbum {
  entries: CockpitAlbumEntry[];
}

export interface CockpitAlbumEntry {
  title: string;
  dateString: string;
  date: Date;
  year: number;
  images: CockpitAlbumImage[];
  previewImage: CockpitAlbumPreviewImage;
}

export interface CockpitAlbumImage {
  meta: {
    title: string;
    asset: string;
  };
  path: string;
}

export interface CockpitAlbumPreviewImage {
  path: string;
}
