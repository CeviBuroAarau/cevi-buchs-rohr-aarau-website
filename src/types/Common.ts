export enum FormState {
  NotDisplayed,
  Displayed,
}
