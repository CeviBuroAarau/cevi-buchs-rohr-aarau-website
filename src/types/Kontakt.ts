export interface KontaktFormRequest {
  form: {
    name: string;
    email: string;
    message: string;
  };
}

export interface KontaktFormResponse {
  error: string;
  data: {
    name: string;
    email: string;
    message: string;
  };
}
