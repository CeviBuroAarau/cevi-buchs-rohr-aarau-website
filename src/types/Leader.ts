export interface Leader {
  name: string;
  isactive: boolean;
  scoutname: string;
  function: string[];
  group: string;
  birthyear: string;
  place: string;
  profession: string;
  recreation: string;
  inScoutsSince: string;
  inScoutsBecause: string;
  bestExperiences: string;
  file: string;
}

export interface LeaderRaw {
  name: string;
  isactive: boolean;
  scoutname: string;
  function: string[];
  group: {
    display: string;
  };
  birthyear: string;
  place: string;
  profession: string;
  recreation: string;
  inScoutsSince: string;
  inScoutsBecause: string;
  bestExperiences: string;
  image: {
    path: string;
  };
}

export interface CockpitLeaders {
  entries: LeaderRaw[];
}
