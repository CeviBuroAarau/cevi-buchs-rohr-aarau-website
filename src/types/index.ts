export * from "./Activity";
export * from "./Agenda";
export * from "./Album";
export * from "./Media";
