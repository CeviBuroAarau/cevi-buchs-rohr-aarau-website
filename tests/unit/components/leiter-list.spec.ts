/* eslint-disable @typescript-eslint/no-explicit-any */
import { mount, shallowMount, Wrapper } from "@vue/test-utils";
import LeiterList from "@/components/leiter-list.vue";
import { Leader } from "@/types";

describe("LeiterList Component", () => {
  const luna: Leader = {
    name: "Leah Dellenbach",
    isactive: true,
    scoutname: "Luna",
    function: ["Gruppenleiterin", "Abteilungsleiterin"],
    group: "Spartacus",
    birthyear: "1998",
    place: "Aarau",
    profession: "Neue Kantonsschule Aarau",
    recreation: "Lesen, Jungschi, mit Freunden treffen",
    inScoutsSince: "ungef\\u00e4hr 2006",
    inScoutsBecause: "es lustig ist mit Gleichgesinnten etwas zu machen",
    bestExperiences: "Pfila 2007, Leiteranl\\u00e4sse",
    file:
      process.env.VUE_APP_COCKPIT_FILES +
      "\\/storage\\/uploads\\/2021\\/02\\/01\\/2017-12-Luna.jpg_uid_6018765b6b961.webp",
  };
  const leiter: Leader[] = [luna, luna];

  test("render leiter-list", () => {
    const wrapper = shallowMount(LeiterList, {
      stubs: ["leiter-detail"],
      propsData: {
        leiter: leiter,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  test("showDetail", async () => {
    const wrapper: Wrapper<LeiterList & { [key: string]: any }> = mount(
      LeiterList,
      {
        propsData: {
          leiter: leiter,
        },
      }
    );

    await wrapper.vm.showDetail(luna);

    const container = wrapper.find("div.is-active");
    expect(container.exists()).toBe(true);
  });
});
