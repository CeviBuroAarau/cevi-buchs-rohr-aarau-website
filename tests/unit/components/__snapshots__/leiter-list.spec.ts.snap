// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeiterList Component render leiter-list 1`] = `
<div>
  <leiter-detail-stub />
   
  <ul
    class=""
    style="display: flex; flex-wrap: wrap;"
  >
    <li
      class="leiterlist"
    >
      <div
        class="card"
        style="width: 180px;"
      >
        <div
          class="card-image is-centered"
        >
          <img
            alt="Luna"
            src="https://cockpit.cevi-buro-aarau.ch/\\\\/storage\\\\/uploads\\\\/2021\\\\/02\\\\/01\\\\/2017-12-Luna.jpg_uid_6018765b6b961.webp"
            width="170"
          />
        </div>
         
        <div
          class="card-content is-centered"
        >
          <strong>
            Leah Dellenbach
            <br />
            
            v/o Luna
          </strong>
        </div>
      </div>
    </li>
    <li
      class="leiterlist"
    >
      <div
        class="card"
        style="width: 180px;"
      >
        <div
          class="card-image is-centered"
        >
          <img
            alt="Luna"
            src="https://cockpit.cevi-buro-aarau.ch/\\\\/storage\\\\/uploads\\\\/2021\\\\/02\\\\/01\\\\/2017-12-Luna.jpg_uid_6018765b6b961.webp"
            width="170"
          />
        </div>
         
        <div
          class="card-content is-centered"
        >
          <strong>
            Leah Dellenbach
            <br />
            
            v/o Luna
          </strong>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
